src/main/java/XiaoLong.java

```diff
@@ -0,0 +1,191 @@
+        }
+    }
+
+    private void handleDeadlineCommand(String input) {
+        try {
+            String[] parts = input.substring(9).split(" /by ");
+            tasks.add(new Deadline(parts[0], parts[1]));
+            saveTasksToFile();
+            System.out.println("Got it. I've added this task:\n  " + tasks.get(tasks.size() - 1));
+        } catch (Exception e) {
+            System.out.println("Invalid Deadline format! Use: deadline [task] /by [time]");
+        }
+    }
+
+    private void handleEventCommand(String input) {
+        try {
+            String[] parts = input.substring(6).split(" /from | /to ");
+            if (parts.length < 3) {
+                throw new IllegalArgumentException();
+            }
+            tasks.add(new Event(parts[0], parts[1], false));
+            saveTasksToFile();
+            System.out.println("Got it. I've added this task:\n  " + tasks.get(tasks.size() - 1));
+        } catch (Exception e) {
+            System.out.println("Invalid Event format! Use: event [task] /from [start] /to [end]");
+        }
+    }
+
+    private void handleDeleteCommand(String input) {
+        try {
+            int taskNumber = Integer.parseInt(input.split(" ")[1]) - 1;
+            Task removedTask = tasks.remove(taskNumber);
+            saveTasksToFile();
+            System.out.println("Noted. I've removed this task:\n  " + removedTask);
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        } catch (Exception e) {
+            System.out.println("Invalid task number! Use: delete [task number]");
+        }
+    }
+
+    private void saveTasksToFile() {
+        try {
+            File file = new File(FILE_PATH);
+            file.getParentFile().mkdirs(); // Ensure the directory exists
+
+            BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH));
+            for (Task task : tasks) {
+                writer.write(task.toFileFormat());
+                writer.newLine();
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error saving tasks to file.");
+        }
+    }
+
+    private void loadTasksFromFile() {
+        File file = new File(FILE_PATH);
+        if (!file.exists()) {
+            return;
+        }
+
+        try {
+            Scanner fileScanner = new Scanner(file);
+            while (fileScanner.hasNextLine()) {
+                String line = fileScanner.nextLine();
+                Task task = Task.fromFileFormat(line);
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+            fileScanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found. Starting with an empty task list.");
+        }
+    }
+
+    public static void main(String[] args) {
+        XiaoLong xiaoLong = new XiaoLong();
```

Comment:
Usage of english words is encouraged in code as it caters to international audience.